app/helpers/compute_resources_vms_helper.rb

```diff
@@ -142,8 +142,12 @@ def show_vm_name?
     controller_name != 'hosts' && controller_name != 'compute_attributes'
   end
 
-  def new_host?(host)
-    host.try(:new_record?)
+  def new_host?(host, options = {})
+    if host.nil? && options[:nil_is_new]
```

Comment:
I'm not exactly sure how to make this work, what did you mean by removing @host? it's definitely needed (the variable), because it's used in the #new or #edit contexts.
but yeah, if f.object isn't a host, the new_host? method makes no sense there, as it's not a host...